File: python/118.py
#time:16ms;memory:13MB
class Solution(object):
    def generate(self, numRows):
        """
        :type numRows: int
        :rtype: List[List[int]]
        """
        yh = []
        for i in range(numRows):
            tlist = []
            for j in range(i+1):
                if j==0 or j==i:
                    tlist.append(1)
                else:
                    tlist.append(yh[i-1][j-1]+yh[i-1][j])
            yh.append(tlist)
        return yh

File: java/118.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> list = new ArrayList<List<Integer>>();
        for(int i = 0;i<numRows;i++){
            ArrayList<Integer> templ = new ArrayList<Integer>(i+1);
            for(int j=0;j<i+1;j++){
                if(i==j || j==0){
                    templ.add(1);
                }else{
                    templ.add(list.get(i-1).get(j)+list.get(i-1).get(j-1));
                }
            }
            list.add(templ);
        }
        return list;
    }
}